### PR TITLE
[ODH-2366]: STB-General, Ethernet/WiFi Reports basic-analytics

### PR DESCRIPTION
**Pull request by EgorAfutin**

### CODE REVIEW

File: src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    def distinct_total_ethernet_network_types_count(self):
+        return self._common_ethernet_pipeline \
+            .where((col("rxKbps") >= 1) | (col("txKbps") >= 1)) \
+            .aggregate(DistinctCount(aggregation_field="viewer_id", group_fields=["type"],
+                                     aggregation_name=self._component_name + ".network_type"))
+
+    # Ethernet Usage Report
+    def ethernet_average_upstream_kbps(self):
+        return self._common_ethernet_pipeline \
+            .where("txKbps is not NULL") \
```
Comment (Aronornone) [inline src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports.py]: Test comment

File: src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
+"""Module for counting all general analytics metrics for EOS STB Ethernet/Wifi Report"""
```
Comment (halfwhite) [inline src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py]: It'll be more readable if it's a multiline comment.

File: src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+from pyspark.sql.types import StructField, StructType, TimestampType, StringType, ArrayType, IntegerType
+
+from common.basic_analytics.basic_analytics_processor import BasicAnalyticsProcessor
+from util.kafka_pipeline_helper import start_basic_analytics_pipeline
+from common.basic_analytics.aggregations import DistinctCount, Avg
+from pyspark.sql.functions import col, explode
+
+
+class EthernetWifiReportEventProcessor(BasicAnalyticsProcessor):
+    """Class that's responsible to process pipelines for Ethernet/Wifi Reports"""
```
Comment (halfwhite) [inline src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py]: Let's create one line comment for fields and multiline for others.

File: src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+    def _process_pipeline(self, read_stream):
+        report_generator = EthernetWifiReports(read_stream, self._component_name)
+
+        return [report_generator.distinct_total_wifi_network_types_count(),
+                report_generator.distinct_total_ethernet_network_types_count(),
+                report_generator.ethernet_average_upstream_kbps(), report_generator.ethernet_average_downstream_kbps(),
```
Comment (halfwhite) [inline src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py]: one invocation on one line is read better.

File: src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        ])
+
+
+class EthernetWifiReports(object):
+    """Class that is able to create output streams with metrics for Ethernet Wifi Reports"""
+
+    def __init__(self, read_stream, component_name):
+        self._component_name = component_name
+
+        # Common Wifi Report
```
Comment (halfwhite) [inline src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py]: Don't use inline comments. Use meaningful variable name.

File: src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+            .select("@timestamp",
+                    explode("NetConfiguration.ifaces").alias("ifaces"),
+                    col("header.viewerID").alias("viewer_id")) \
+            .select("@timestamp",
+                    col("ifaces.enabled").alias("net_config_enabled"),
+                    col("ifaces.type").alias("net_configuration_type"),
+                    col("viewer_id"))
+
+    # STB Network Type
+    def distinct_total_wifi_network_types_count(self):
```
Comment (halfwhite) [inline src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py]: It's a method. Use string doc.

File: src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        return self._common_wifi_pipeline \
+            .where((col("rxKbps") >= 1) | (col("txKbps") >= 1)) \
+            .aggregate(DistinctCount(aggregation_field="viewer_id", group_fields=["type"],
+                                     aggregation_name=self._component_name + ".network_type"))
+
+    def distinct_total_ethernet_network_types_count(self):
+        return self._common_ethernet_pipeline \
```
Comment (halfwhite) [inline src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py]: It looks metric names will be the same.

File: src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+            .where("rxKbps is not NULL") \
+            .aggregate(Avg(aggregation_field="rxKbps",
+                           aggregation_name=self._component_name + ".wireless.average_downstream_kbps"))
+
+    # NetConfig Enabled/Disabled STB's
+    def distinct_total_net_config_enabled(self):
+        return self._common_net_configuration_pipeline \
+            .where("net_config_enabled is not NULL") \
+            .aggregate(DistinctCount(aggregation_field="viewer_id", group_fields=["net_config_enabled"],
+                                     aggregation_name=self._component_name))
```
Comment (halfwhite) [inline src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py]: Any meaningful suffix?

File: src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+    # Total CPE NetConfig For Wifi/Ethernet Channels
+    def total_cpe_net_config_for_wifi_ethernet_channels(self):
+        return self._common_net_configuration_pipeline \
+            .where("net_configuration_type is not NULL") \
+            .aggregate(DistinctCount(aggregation_field="viewer_id", group_fields=["net_configuration_type"],
+                                     aggregation_name=self._component_name))
```
Comment (halfwhite) [inline src/applications/basic_analytics/stb_general/cpe_ethernet_wifi_reports_driver.py]: Any meaningful suffix?

**Pull request merged**